lib/DBusProxy.js

```diff
@@ -571,34 +586,37 @@ DBusProxy.prototype.makeIntrospectionPass = function (introspectionDepth, target
 										Same deal as for 'PropertiesChanged', we catch it here so that the user cannot
 										accidently remove it.
 									*/
-									if (signalName === 'InterfacesAdded' && this.isInterfaceUpdatingAllowed) {
+									if (signalName === 'InterfacesAdded') {
 										let objPath = translatedArgs[0]
-										let currPath = '/' + parent.getPath() + '/'
-										let targetPath = objPath.replace(currPath, '')
-										let pathComponents = targetPath.split('/')
-										let currObj = parent
-
-										while (pathComponents.length > 0) {
-											let pathComponent = pathComponents.shift()
+										let depth = this.maxIntrospectionDepth - this.getDepth(objPath) + 1
+										if (depth > 0 && (this.pathsAreIncluded(targetedPath, objPath))) {
+											let currPath = '/' + parent.getPath() + '/'
+											let targetPath = objPath.replace(currPath, '')
+											let pathComponents = targetPath.split('/')
+											let currObj = parent
+
+											while (pathComponents.length > 0) {
+												let pathComponent = pathComponents.shift()
+
+												if (currObj[pathComponent] === undefined) {
+													currObj.addObject(new DBusObject(), pathComponent)
+												}
 
-											if (currObj[pathComponent] === undefined) {
-												currObj.addObject(new DBusObject(), pathComponent)
+												currObj = currObj[pathComponent]
 											}
 
-											currObj = currObj[pathComponent]
+											debug(`'InterfacesAdded' or 'InterfacesRemoved' detected. Making introspection pass at '${objPath}, depth: ${depth}`)
+											this.makeIntrospectionPass(depth, currObj, objPath, targetedPath, targetedIface)
+											.then( _ => {
+												debug(`done, this:\n${inspect(this, {depth: 7})}`)
+											})
 										}
-
-										debug(`'InterfacesAdded' or 'InterfacesRemoved' detected. Making introspection pass at '${objPath}'`)
-										this.makeIntrospectionPass(this.maxIntrospectionDepth, targetedPath, currObj, objPath)
-										.then( _ => {
-											debug(`done, this:\n${inspect(this, {depth: 7})}`)
-										})
 									}
 
 									/*
 										Handle 'InterfacesRemoved' signal.
 									*/
-									if (signalName === 'InterfacesRemoved' && this.isInterfaceUpdatingAllowed) {
+									if (signalName === 'InterfacesRemoved') {
```

Comment:
What if an interface that was not managed is removed ? Does it make the library crash ?
Have you tested ?

Reply:
The change seems ok with respect to this question. I assume, you have tested.